src/core/options-watcher.cpp

```diff
@@ -18,8 +18,10 @@ options_watcher::~options_watcher()
 {
     try
     {
-        std::lock_guard< std::mutex > lock( _mutex );
-        _options.clear();
+        {
+            std::lock_guard< std::mutex > lock( _mutex );
```

Comment:
This will lock, so `should_stop()` will never return true inside `update_options()`... right?

Reply:
Only if subscription is canceled.

If destructor is called in the middle of the queries it will wait till all the queries have completed

Reply:
Added a flag to stop queries as soon as destructor is called

Reply:
Why not do `_alive = false` when stopping?

Reply:
I don't want this a a thread running flag, it complicates `should_start/stop` with races of changing the flag value and actual thread starting/stoping.
I am using this flag for destruction signaling only, to stop querying as soon as possible. Renamed the flag `_destructing`